week-3/refactor/format-as-12-hours.js

```diff
@@ -4,3 +4,61 @@
 // Store this expression in a variable and reference it twice in the function in the correct place
 
 // Explain why it makes more sense to store this expression in a variable
+
+// Given code in the file in debug folder:
+
+// function formatAs12HourClock(time) {
+//   if (Number(time.slice(0, 2)) > 12) {
+//     return `${Number(time.slice(0, 2)) - 12}:00 pm`;
+//   }
+//   return `${time} am`;
+// }
+
+// const currentOutput = formatAs12HourClock("08:00");
+// const targetOutput = "08:00 am";
+// console.assert(
+//   currentOutput === targetOutput,
+//   "current output: %s, target output: %s",
+//   currentOutput,
+//   targetOutput
+// );
+
+// const currentOutput2 = formatAs12HourClock("23:00");
+// const targetOutput2 = "11:00 pm";
+// console.assert(
+//   currentOutput2 === targetOutput2,
+//   "current output: %s, target output: %s",
+//   currentOutput2,
+//   targetOutput2
+// );
+
+// Improved and fixed version of it:
+
+function formatAs12HourClock(time) {
+  const hours = Number(time.slice(0, 2));
+  const minutes = Number(time.slice(3));
+
+  if (hours >= 12) {
+    const formattedHoursAfternoon =
```

Comment:
there is a concept called Do not Repeat Yourself (DRY) it is a concept where you write a code once and you reuse it, instead of rewritten the same code again, you can read more about. `formattedHoursAfternoon` and `formattedHoursMorning` have a couple of common code, can you think of a way to extract that into a reusable function?